src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,35 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                let check_lock_res =
+                    self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock));
```

Comment:
Maybe better to use:

```rust
let check_lock_res = self.load_and_check_lock(
    user_key, 
    can_be_cached.as_ref().map(|_| &mut can_be_cached_by_lock)
);
```

so that `load_and_check_lock()` receives `load_and_check_lock(.., None)` when `get()` receives `get(..., None)`.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=ce34cc3b4925da7e012560445b7bb858